app/forms/pet_form.rb

```diff
@@ -0,0 +1,41 @@
+class PetForm
+  include ActiveModel::Model
+
+  attr_accessor :photos
+
+  validates :photos, presence: true
+
+  validate :validates_number_of_files
+  # FILE_NUMBER_LIMIT = 4 #枚数制限（最大数）
+
+  def validates_number_of_files
+    binding.pry
+
+    # 枚数に関係なくひかかる、保存されない
+    @photos.present? && @photos.length < 5
+      errors.add(:@photos, '添付枚数は４枚までです。')
```

Comment:
普通に見落としてしまってましたが↓ですね。

```suggestion
    if photos.present? && photos.length > 4
      errors.add(:photos, '添付枚数は４枚までです。')
    end
```

こちらに修正するとどうでしょう？